src/caffe/parallel.cpp

```diff
@@ -82,14 +83,17 @@ GPUParams<Dtype>::GPUParams(shared_ptr<Solver<Dtype> > root_solver, int device)
 
   // Allocate device buffers
   CUDA_CHECK(cudaSetDevice(device));
-  CUDA_CHECK(cudaMalloc(&data_, size_ * sizeof(Dtype)));
+  buffer_device_ = device;
+  GPUMemoryManager::allocate(reinterpret_cast<void **>(&data_),
```

Comment:
src/caffe/parallel.cpp:95:52: error: invalid conversion from ‘float**’ to ‘void**’ [-fpermissive]
   GPUMemory::allocate(&diff_, size_ * sizeof(Dtype));
                                                    ^
In file included from src/caffe/parallel.cpp:14:0:
./include/caffe/util/gpu_memory.hpp:16:15: error:   initializing argument 1 of ‘static void caffe::GPUMemory::allocate(void**, size_t, cudaStream_t)’ [-fpe\
rmissive]
   static void allocate(void** ptr, size_t size,
               ^